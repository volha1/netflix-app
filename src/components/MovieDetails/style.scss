@import '../../common/scss/base.scss';

.movie-details {
  padding: 35px 0px;

  .nav-block {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 30px;
  }

  .search-icon {
    cursor: pointer;
  }

  .details-wrapper {
    display: flex;
  }

  .movie-img {
    display: block;
    width: 323px;
    height: 486px;
    margin-right: 60px;
  }

  .title-wrapper {
    display: flex;

    .title {
      margin-bottom: 10px;
    }

    .rating {
      border: 1px solid $white;
      border-radius: 50%;
      width: auto;
      display: inline-block;
      font-size: 20px;
      line-height: 40px;
      vertical-align: 7px;
      margin-left: 40px;
      height: 40px;
      width: 40px;
      text-align: center;
    }
  }

  .genres {
    opacity: 0.5;
    margin-bottom: 30px;
  }

  .movie-characteristics {
    color: $red;
    font-size: 24px;
    font-weight: 300;
    margin-bottom: 30px;

    span {
      margin-right: 30px;
    }
  }

  .overview {
    opacity: 0.5;
    font-weight: 300;
    font-size: 20px;
    line-height: 24px;
  }
}
