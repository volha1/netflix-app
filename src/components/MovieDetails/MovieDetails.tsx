import React, { ReactElement, useCallback } from 'react';
import searchIcon from '../../common/assets/svg/search-icon.svg';
import Movie from '../../entity/Movie';
import { getMovieDuration, getYear } from '../../helpers/utils';
import setDefaultImage from '../../helpers/setDefaultImage';
import './style.scss';

type MovieDetailsProps = {
  movie: Movie | null;
  onSelectMovie: (value: null) => void;
};

const MovieDetails = ({ movie, onSelectMovie }: MovieDetailsProps): ReactElement | null => {
  const handleSeacrhIconClick = useCallback((): void => {
    onSelectMovie(null);
  }, []);

  return (
    movie && (
      <div className="movie-details">
        <div className="movie-details-wrapper content">
          <div className="nav-block">
            <p className="logo">netflixroulette</p>
            <img className="search-icon" src={searchIcon} alt="Search" onClick={handleSeacrhIconClick} />
          </div>
          <div className="details-wrapper">
            <img className="movie-img" src={movie.imgPath} alt={movie.title} onError={setDefaultImage} />
            <div className="movie-description">
              <div className="title-wrapper">
                <h1 className="title">
                  {movie.title}
                  <span className="rating">{movie.voteAverage}</span>
                </h1>
              </div>
              <p className="genres">{movie.genres.join(', ')}</p>
              <p className="movie-characteristics">
                <span>{getYear(movie.year)}</span>
                {getMovieDuration(movie.runtime)}
              </p>
              <p className="overview">{movie.overview}</p>
            </div>
          </div>
        </div>
      </div>
    )
  );
};

export default MovieDetails;
