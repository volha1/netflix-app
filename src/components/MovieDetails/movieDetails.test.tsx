import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as reduxHooks from 'react-redux';

import MovieDetails from './MovieDetails';

jest.mock('react-redux');

describe('MovieDetails test', () => {
  const removeSearchParams = jest.fn();
  jest.spyOn(reduxHooks, 'useSelector').mockReturnValue([]);
  jest.spyOn(React, 'useMemo').mockReturnValue({
    id: '1',
    title: 'title',
    voteAverage: '10',
    releaseDate: '2022',
    imgPath: 'https://test.com',
    genres: ['Horror'],
    overview: 'overview text',
    runtime: 110,
  });

  it('MovieDetails snapshot', () => {
    const component = renderer.create(<MovieDetails removeSearchParams={removeSearchParams} movieId="" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('removeSearchParams function works', async () => {
    render(<MovieDetails removeSearchParams={removeSearchParams} movieId="" />);

    await userEvent.click(screen.getByTestId('search-icon'));

    expect(removeSearchParams).toHaveBeenCalledWith('movie');
  });
});
