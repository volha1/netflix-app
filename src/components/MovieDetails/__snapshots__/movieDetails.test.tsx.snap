// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails test MovieDetails snapshot 1`] = `
<div
  className="movie-details"
>
  <div
    className="movie-details-wrapper content"
  >
    <div
      className="nav-block"
    >
      <p
        className="logo"
      >
        netflixroulette
      </p>
      <img
        alt="Search"
        className="search-icon"
        data-testid="search-icon"
        onClick={[Function]}
      />
    </div>
    <div
      className="details-wrapper"
    >
      <img
        alt="title"
        className="movie-img"
        onError={[Function]}
        src="https://test.com"
      />
      <div
        className="movie-description"
      >
        <div
          className="title-wrapper"
        >
          <h1
            className="title"
          >
            title
            <span
              className="rating"
            >
              10
            </span>
          </h1>
        </div>
        <p
          className="genres"
        >
          Horror
        </p>
        <p
          className="movie-characteristics"
        >
          <span>
            2022
          </span>
          1h 50m
        </p>
        <p
          className="overview"
        >
          overview text
        </p>
      </div>
    </div>
  </div>
</div>
`;
